### DISCUSSION TITLE
Regex search and more ..

### DISCUSSION THREAD

**Discussion by vsraptor** [KeyviDev/keyvi#230] [Ideas]
Similar project can do regex search, Set operations and more : 

https://blog.burntsushi.net/transducers/#regular-expressions
https://blog.burntsushi.net/transducers/#set-operations
https://blog.burntsushi.net/transducers/#levenshtein-automata

**Comment by hendrikmuhs** [comment]
Can we move this to Discussions?

We are aware of this implementation, it has some nice features indeed. Note that this comes back to the index vs. dictionary discussion. Afaik this is a immutable implementation, so it is a dictionary, however e.g. the tantivy search engine build the index concept on top of that.

keyvi has fuzzy matching(e.g. levenshtein) support, too, even in the index. I experimented with using the FST implementation for regular expressions, but it never made it to keyvi.

Implementationwise this implementation is close to the one in Lucene, it uses linked lists afaik, keyvi uses sparse arrays. In my opinion this overcomes some traversal overhead, well there are pro's and con's for every implementation.

Happy to continue chatting, but as said, lets move this to discussions and keep "issues" for bugs/enhancement requests.

**Comment by vsraptor** [comment]
how do i move it to discussions
